### ISSUE TITLE
Error When Creating a new Region 7680x7680 or 8192x8192

### ISSUE THREAD

**Issue by TommyFDNY** [WhiteCoreSim/WhiteCore-Dev#124]
Jan 15 12:49:59 - [Console]: Console log level is Info
Jan 15 12:49:59 - ==============================================================
======
Jan 15 12:49:59 - ==================== STARTING WhiteCore (x86) ================
======
Jan 15 12:49:59 - ==============================================================
======
Jan 15 12:49:59 - [WhiteCoreStartup]: Version: WhiteCore 0.9.3 Dev (2014-12-05.7
5eacf9)
Jan 15 12:49:59 - [Network]: Retrieving the external IP address
Jan 15 12:49:59 - [Network]: Network IP address has been set to 8.38.77.92
Jan 15 12:49:59 - [BASE HTTP SERVER]: Starting HTTP server on port 9000
Jan 15 12:49:59 - [VivoxVoice] using vivox server www.osp.vivox.com
Jan 15 12:49:59 - [VivoxVoice] Admin connection established
Jan 15 12:49:59 - [VivoxVoice]: plugin enabled
Jan 15 12:49:59 - [SMTP] Email enabled for Local only
Jan 15 12:50:00 - Looking for previous sims in: ../Data/Region
Jan 15 12:50:00 - Found: Pularas Immigration Island
Jan 15 12:50:00 - [FileBasedSimulationData]: Restoring sim backup for region Pul
aras Immigration Island...
Jan 15 12:50:00 - [SceneManager]: Starting region "Pularas Immigration Island" a
t @ 1030,1030
Jan 15 12:50:00 - [MaterialsDemoModule]: Initializing module
Jan 15 12:50:00 - [WORLDVIEW]: Configured and enabled for Pularas Immigration Is
land
Jan 15 12:50:00 - [WORLDVIEW]: RegionID 2bf6d012-0f59-49a9-b6c8-02b3a21237e1
Jan 15 12:50:00 - [BackupModule]: Loading objects for Pularas Immigration Island
 from FileBasedDatabase
Jan 15 12:50:01 - [BackupModule]: Loaded 96 object(s) in Pularas Immigration Isl
and
Jan 15 12:50:01 - [BackupModule]: Starting scripts in Pularas Immigration Island

Jan 15 12:50:01 - [SceneManager]: Startup Complete. This took 0m 1.684s
root: # create region
Region Name (? for suggestion) [Thlonder]: Pularas
Region Location X [1031]: 1040
Region location Y [1031]: 1040
Region size [256]: 7680
Region Type (Mainland/Estate) [Estate]:
Estate region type?, Options are [Full Region, Whitecore, Custom] [Full Region]:
 Whitecore
Jan 15 12:52:52 - [SceneManager]: Starting region "Pularas" at @ 1040,1040
Jan 15 12:52:52 - [MaterialsDemoModule]: Initializing module
Jan 15 12:52:52 - [TERRAIN]: No default terrain. Generating a new terrain.
Jan 15 12:53:09 - [TERRAIN]: Scene.cs: LoadWorldMap() - Failed with exception Sy
stem.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was t
hrown.
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GetEmptyArray[T](Int32 width, In
t32 height) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainPer
lin.cs:line 121
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GenerateSmoothNoise(Single[][] b
aseNoise, Int32 octave) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInf
o\TerrainPerlin.cs:line 132
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GeneratePerlinNoise(Single[][] b
aseNoise, Int32 octaveCount) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\Sce
neInfo\TerrainPerlin.cs:line 179
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GeneratePerlinNoise(Int32 width,
 Int32 height, Int32 octaveCount) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framewor
k\SceneInfo\TerrainPerlin.cs:line 219
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GenerateHeightMap(Int32 width, I
nt32 height, Int32 octaveCount, Single min, Single max, Int32 smoothing) in e:\G
itLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainPerlin.cs:line 655
   at WhiteCore.Framework.SceneInfo.TerrainChannel.CreateMainlandTerrain(Single
minHeight, Single maxHeight, Int32 smoothing) in e:\GitLabHQ\WhiteCoreDev\WhiteC
ore\Framework\SceneInfo\TerrainChannel.cs:line 401
   at WhiteCore.Framework.SceneInfo.TerrainChannel.CreateMainlandTerrain(Int32 s
moothing) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainChann
el.cs:line 383
   at WhiteCore.Framework.SceneInfo.TerrainChannel.CreateDefaultTerrain(String l
andType) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainChanne
l.cs:line 312
   at WhiteCore.Framework.SceneInfo.TerrainChannel..ctor(IScene scene) in e:\Git
LabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainChannel.cs:line 54
   at WhiteCore.Modules.Terrain.TerrainModule.LoadWorldHeightmap() in e:\GitLabH
Q\WhiteCoreDev\WhiteCore\Modules\World\Terrain\TerrainModule.cs:line 298
Jan 15 12:53:26 - [RegionModulesControllerPlugin]: Failed to load module Terrain
Module: System.OutOfMemoryException: Exception of type 'System.OutOfMemoryExcept
ion' was thrown.
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GetEmptyArray[T](Int32 width, In
t32 height) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainPer
lin.cs:line 121
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GenerateSmoothNoise(Single[][] b
aseNoise, Int32 octave) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInf
o\TerrainPerlin.cs:line 132
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GeneratePerlinNoise(Single[][] b
aseNoise, Int32 octaveCount) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\Sce
neInfo\TerrainPerlin.cs:line 179
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GeneratePerlinNoise(Int32 width,
 Int32 height, Int32 octaveCount) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framewor
k\SceneInfo\TerrainPerlin.cs:line 219
   at WhiteCore.Framework.SceneInfo.PerlinNoise.GenerateHeightMap(Int32 width, I
nt32 height, Int32 octaveCount, Single min, Single max, Int32 smoothing) in e:\G
itLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainPerlin.cs:line 655
   at WhiteCore.Framework.SceneInfo.TerrainChannel.CreateMainlandTerrain(Single
minHeight, Single maxHeight, Int32 smoothing) in e:\GitLabHQ\WhiteCoreDev\WhiteC
ore\Framework\SceneInfo\TerrainChannel.cs:line 401
   at WhiteCore.Framework.SceneInfo.TerrainChannel.CreateMainlandTerrain(Int32 s
moothing) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainChann
el.cs:line 383
   at WhiteCore.Framework.SceneInfo.TerrainChannel.CreateDefaultTerrain(String l
andType) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainChanne
l.cs:line 312
   at WhiteCore.Framework.SceneInfo.TerrainChannel..ctor(IScene scene) in e:\Git
LabHQ\WhiteCoreDev\WhiteCore\Framework\SceneInfo\TerrainChannel.cs:line 54
   at WhiteCore.Modules.Terrain.TerrainModule.LoadWorldHeightmap() in e:\GitLabH
Q\WhiteCoreDev\WhiteCore\Modules\World\Terrain\TerrainModule.cs:line 332
   at WhiteCore.Modules.Terrain.TerrainModule.AddRegion(IScene scene) in e:\GitL
abHQ\WhiteCoreDev\WhiteCore\Modules\World\Terrain\TerrainModule.cs:line 134
   at WhiteCore.CoreApplicationPlugins.RegionModulesController.RegionModulesCont
rollerPlugin.AddRegionToModules(IScene scene) in e:\GitLabHQ\WhiteCoreDev\WhiteC
ore\CoreApplicationPlugins\RegionModulesController\RegionModulesControllerPlugin
.cs:line 121
Jan 15 12:53:26 - [WORLDVIEW]: Configured and enabled for Pularas
Jan 15 12:53:26 - [WORLDVIEW]: RegionID a335cb9c-87bd-471f-b953-edb6ccb3586d
Jan 15 12:53:26 - [EstateInitializer]: Your region 'Pularas' is not part of an e
state.
Estate owner name (RealEstate Owner/User Name) [RealEstate Owner]:
Jan 15 12:54:03 - [Estate]: Successfully joined the Whitecore Estate estate!
Jan 15 12:54:03 - [BackupModule]: Loading objects for Pularas from FileBasedData
base
Jan 15 12:54:03 - [BackupModule]: Loaded 0 object(s) in Pularas
Jan 15 12:54:03 - [ParcelManagement]: Setting land owner for region Pularas to R
ealEstate Owner
Jan 15 12:54:03 - [BackupModule]: Starting scripts in Pularas
Jan 15 12:54:22 - [SceneManager]: Startup Complete. This took 4m 22.956s
Jan 15 12:54:22 - [FileBasedSimulationData]: Backing up Pularas Immigration Isla
nd
Jan 15 12:54:22 - [FileBasedSimulationData]: Saved Backup for region Pularas Imm
igration Island
Jan 15 12:54:22 - [FileBasedSimulationData]: Backing up Pularas
Jan 15 12:54:22 - [Backup]: Exception caught: System.NullReferenceException: Obj
ect reference not set to an instance of an object.
   at WhiteCore.Modules.FileBasedSimulationData.WriteTerrainToStream(ITerrainCha
nnel tModule) in e:\GitLabHQ\WhiteCoreDev\WhiteCore\Modules\World\Startup\Simula
tionData\FileBasedSimulationData.cs:line 1063
   at WhiteCore.Modules.FileBasedSimulationData.SaveBackup(Boolean isOldSave) in
 e:\GitLabHQ\WhiteCoreDev\WhiteCore\Modules\World\Startup\SimulationData\FileBas
edSimulationData.cs:line 962
Jan 15 12:54:22 - [FileBasedSimulationData]: Saved Backup for region Pularas
root

**Comment by TommyFDNY** [comment]
Flyman mentioned that this is a past bug that he submitted, and it was fixed; it looks like it resurfaced.

**Comment by fly-man-** [comment]
@TommyFDNY It seems that making regions larger then 4096 overburden the region generation. I think @greythane did a quick fix for it but not sure what happened after it.

**Comment by TommyFDNY** [comment]
Ok strange it was working for days, than all of a sudden. I tried deleting the region and re creating :-(

**Comment by TommyFDNY** [comment]
Re-Created Region in advised 4096 size, no errors. is happy for now.

**Comment by greythane** [comment]
Terrain generation does suck up memory during creation.
I have verified many times that a region of 8192x8192 (current max siz) can be created  OS X with 4gb.  Of course windows and differing amounts of ram will effect this.

Creating a 'flatland' terrain and then loading your terrain height map from jpg, png etc does not go through the intensive memory steps.

**Comment by TommyFDNY** [comment]
Thats strange, I am on a 32gig or ram, 16core server. Everything was running great, all of sudden... whammo.

**Comment by TommyFDNY** [comment]
I have attempted to recreate the 8192x8192 region to no avail. so I down sized to 30x256 sims that was 7680x7680. Still getting the errors, so I downsized to 4096x4096, now it works.

**Comment by TommyFDNY** [comment]
This still may be a bug; I should have mentioned that I was messing around with my MaxThreadPool settings. I multiplied 15 x my 16 core CPU ( I set it to 190), and that is when it got angry at me. 

If I recall that when calculating MaxThreadPool, little's law came into play, so I went back and calculated, and made it around 90 instead of the the 190, or the default for a quad core at 45.

I will close this so you don't kill me, but I think the Sarge mentioned this was a bug that recived a quick fix. THANK YOU!!! I am learning!!
